src/main/scala/beam/agentsim/infrastructure/taz/TAZTreeMap.scala

```diff
@@ -67,6 +80,39 @@ class TAZTreeMap(val tazQuadTree: QuadTree[TAZ], val useCache: Boolean = false)
   def getTAZInRadius(loc: Coord, radius: Double): util.Collection[TAZ] = {
     tazQuadTree.getDisk(loc.getX, loc.getY, radius)
   }
+
+  def mapNetworkToTAZs(network: Network): Unit = {
+    if (tazListContainsGeoms) {
+      network.getLinks.asScala.foreach {
+        case (id, link) =>
+          val linkEndCoord = link.getToNode.getCoord
+          val foundTaz = TAZTreeMap.ringSearch(
+            tazQuadTree,
+            linkEndCoord,
+            100,
+            1000000,
+            radiusMultiplication = 1.5
+          ) { taz =>
+            if (taz.geometry.exists(_.contains(GeometryUtils.createGeotoolsPoint(linkEndCoord)))) { Some(taz) }
+            else None
+          }
+          foundTaz match {
+            case Some(taz) =>
+              linkIdToTAZMapping += (id -> taz.tazId)
+            case _ =>
+              unmatchedLinkIds += id
+          }
+        case _ =>
+      }
+      logger.info(
+        "Matched "
+        + linkIdToTAZMapping.size.toString +
+        " links to TAZs, failed to match "
+        + unmatchedLinkIds.size.toString +
+        " links"
+      )
+    }
+  }
```

Comment:
How long will it take to process a big map such as sf-bay or new-york?
If it might take more than 10 minutes - let's change the log text by adding there the word 'Completed' for 'beam stuck guard'.